feat: remove s3 file from annotations and use an observation array in…

Nastaliss opened the pull request:
…stead

Nastaliss committed:
feat: remove s3 file from annotations and use an observation array instead

ghost commented:
Nice of you to open a PR :pray:
When you're ready and want to get it reviewed, post a comment in this Pull Request with this message: `/quack review`

dependabot[bot] committed:
chore(deps): Bump certifi from 2022.12.7 to 2023.7.22 (#29)

Bumps [certifi](https://github.com/certifi/python-certifi) from 2022.12.7 to 2023.7.22.
- [Commits](https://github.com/certifi/python-certifi/compare/2022.12.07...2023.07.22)

---
updated-dependencies:
- dependency-name: certifi
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump gitpython from 3.1.31 to 3.1.34 (#31)

Bumps [gitpython](https://github.com/gitpython-developers/GitPython) from 3.1.31 to 3.1.34.
- [Release notes](https://github.com/gitpython-developers/GitPython/releases)
- [Changelog](https://github.com/gitpython-developers/GitPython/blob/main/CHANGES)
- [Commits](https://github.com/gitpython-developers/GitPython/compare/3.1.31...3.1.34)

---
updated-dependencies:
- dependency-name: gitpython
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump gitpython from 3.1.34 to 3.1.35 (#32)

Bumps [gitpython](https://github.com/gitpython-developers/GitPython) from 3.1.34 to 3.1.35.
- [Release notes](https://github.com/gitpython-developers/GitPython/releases)
- [Changelog](https://github.com/gitpython-developers/GitPython/blob/main/CHANGES)
- [Commits](https://github.com/gitpython-developers/GitPython/compare/3.1.34...3.1.35)

---
updated-dependencies:
- dependency-name: gitpython
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

frgfm committed:
feat: Enables CORS using FastAPI middleware (#26)

* feat: Adds CORS middleware

* chore: Updates docker

* docs: Updates README

frgfm commented:
Hey there!

Quick question: instead of an array, don't you think we should have a string (JSON dump of the dict)?

dependabot[bot] committed:
chore(deps): Bump urllib3 from 1.26.15 to 1.26.17 (#33)

Bumps [urllib3](https://github.com/urllib3/urllib3) from 1.26.15 to 1.26.17.
- [Release notes](https://github.com/urllib3/urllib3/releases)
- [Changelog](https://github.com/urllib3/urllib3/blob/main/CHANGES.rst)
- [Commits](https://github.com/urllib3/urllib3/compare/1.26.15...1.26.17)

---
updated-dependencies:
- dependency-name: urllib3
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump gitpython from 3.1.35 to 3.1.37 (#34)

Bumps [gitpython](https://github.com/gitpython-developers/GitPython) from 3.1.35 to 3.1.37.
- [Release notes](https://github.com/gitpython-developers/GitPython/releases)
- [Changelog](https://github.com/gitpython-developers/GitPython/blob/main/CHANGES)
- [Commits](https://github.com/gitpython-developers/GitPython/compare/3.1.35...3.1.37)

---
updated-dependencies:
- dependency-name: gitpython
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): Bump urllib3 from 1.26.17 to 1.26.18 (#35)

Bumps [urllib3](https://github.com/urllib3/urllib3) from 1.26.17 to 1.26.18.
- [Release notes](https://github.com/urllib3/urllib3/releases)
- [Changelog](https://github.com/urllib3/urllib3/blob/main/CHANGES.rst)
- [Commits](https://github.com/urllib3/urllib3/compare/1.26.17...1.26.18)

---
updated-dependencies:
- dependency-name: urllib3
  dependency-type: indirect
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Nastaliss commented:
> Hey there!
> 
> Quick question: instead of an array, don't you think we should have a string (JSON dump of the dict)?

I have no preference, if using a jsonb is closer to what's in other tables of the db then sure :)

Nastaliss committed:
fix: image path

Nastaliss committed:
feat: add correct tests for annotations and observations

Nastaliss committed:
feat: sort types

Nastaliss committed:
feat: update only observations

Nastaliss committed:
test: build docker wait for probing doc

Nastaliss committed:
feat: add health-check

Nastaliss committed:
feat: remove hc on backend;

Nastaliss committed:
feat: hard-code env vars for db

Nastaliss committed:
feat: wait for condition healthy

Nastaliss committed:
feat: add health-checks in dev

Nastaliss committed:
feat: lint

frgfm reviewed (commented):
The tests seem to fail with list not being hashable, would you mind checking? 🙏

Nastaliss committed:
chore: retrigger ci

Nastaliss committed:
fix: use string as json key

Nastaliss committed:
fix: no explicit none

Nastaliss committed:
Merge branch 'main' into simplify-annotations

Nastaliss committed:
Merge remote-tracking branch 'origin/main' into simplify-annotations
